src/dodal/devices/i19/pin_col_stages.py

```diff
@@ -0,0 +1,169 @@
+from pydantic import BaseModel
+
+from dodal.devices.i19.mapt_configuration import (
+    MAPTConfigurationControl,
+    MAPTConfigurationTable,
+)
+from dodal.devices.motors import XYStage
+from dodal.log import LOGGER
+
+_PIN = "-MO-PIN-01:"
+_COL = "-MO-COL-01:"
+_CONFIG = "-OP-PCOL-01:"
+
+
+class PinColRequest(StrictEnum):
+    """Aperture request IN positions."""
+
+    PCOL20 = "20um"
+    PCOL40 = "40um"
+    PCOL100 = "100um"
+    PCOL3000 = "3000um"
+    OUT = "OUT"
+
+
+# NOTE. Using subset enum because from the OUT positions should only be used by
+# the beamline scientists from the synoptic. Another option will be needed in the
+# device for OUT position.
+class PinColPosition(SubsetEnum):
+    """Aperture request IN positions."""
+
+    PCOL20 = "20um"
+    PCOL40 = "40um"
+    PCOL100 = "100um"
+    PCOL3000 = "3000um"
+
+
+class AperturePosition(BaseModel):
+    """Describe the positions of the pinhole and collimator stage motors for
+    one of the available apertures.
+
+    Attributes:
+        pinhole_x: The position of the x motor on the pinhole stage
+        pinhole_y: The position of the y motor on the pinhole stage
+        collimator_x: The position of the x motor on the collimator stage
+        collimator_y: The position of the y motor on the collimator stage
+    """
+
+    pinhole_x: float
+    pinhole_y: float
+    collimator_x: float
+    collimator_y: float
+
+
+class PinColConfiguration(StandardReadable):
+    """Full MAPT configuration table, including out positions and selection for the
+    Pinhole and Collimator control."""
+
+    def __init__(self, prefix: str, apertures: list[int], name: str = "") -> None:
+        with self.add_children_as_readables():
+            self.configuration = MAPTConfigurationControl(prefix, PinColPosition)
+            self.pin_x = MAPTConfigurationTable(prefix, "PINX", apertures)
+            self.pin_y = MAPTConfigurationTable(prefix, "PINY", apertures)
+            self.col_x = MAPTConfigurationTable(prefix, "COLX", apertures)
+            self.col_y = MAPTConfigurationTable(prefix, "COLY", apertures)
+            self.pin_x_out = epics_signal_r(float, f"{prefix}:OUT:PINX")
+            self.col_x_out = epics_signal_r(float, f"{prefix}:OUT:COLX")
+        super().__init__(name)
+
+
+class PinholeCollimatorControl(StandardReadable, Movable[str]):
+    """Device to control the Pinhole and Collimator stages moves on I19-2, using the
+    MAPT configuration table to look up the positions."""
+
+    def __init__(
+        self,
+        prefix: str,
+        name: str = "",
+        pin_infix: str = _PIN,
+        col_infix: str = _COL,
+        config_infix: str = _CONFIG,
+    ):
+        self._aperture_sizes = [self._get_aperture_size(i) for i in PinColPosition]
+        with self.add_children_as_readables():
+            self.pinhole = XYStage(f"{prefix}{pin_infix}")
+            self.collimator = XYStage(f"{prefix}{col_infix}")
+            self.mapt = PinColConfiguration(
```

Comment:
Should: I would make all these private so a user isn't tempted to get round the collision detection bit